sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ResourceId.java

```diff
@@ -180,21 +180,21 @@ public static ResourceId newAttachmentId(String documentId, int attachmentId) {
                         ResourceId.blockCopy(buffer, 8, subCollRes, 0, 8);
 
                         long subCollectionResource = ByteBuffer.wrap(buffer, 8, 8).getLong();
-                        if ((subCollRes[7] >> 4) == (byte) CollectionChildResourceType.Document) {
```

Comment:
I am really wondering, why do we need to change the implementation package classes ?
@anuchandy @JonathanGiles  - This will really affect the scope of this PR, as any change in implementation package needs a lot of extra testing and perf runs. It is very difficult to figure our and solve bugs in implementation package. 
Why can't we skip this warnings from implementation package ?
@moderakh - your thoughts ?

Reply:
+1